worksheet.rb

```diff
@@ -1,34 +1,64 @@
-########################################################
-# Step 1: Establish the layers
+require 'csv'
+table = CSV.parse(File.read("rides.csv"), headers: true)
 
-# In this section of the file, as a series of comments,
-# create a list of the layers you identify.
-# Which layers are nested in each other?
-# Which layers of data "have" within it a different layer?
-# Which layers are "next" to each other?
+ride_share = {}
 
+table.each do |row|
+  drive_id = row["DRIVER_ID"]
+  if !ride_share.has_key?(drive_id)
+    ride_share[drive_id] = []
+  end
+  ride_share[drive_id] << {date: row["DATE"], cost: row["COST"].to_i, rider_id: row["RIDER_ID"], rating: row["RATING"].to_i}
+end
 ########################################################
-# Step 2: Assign a data structure to each layer
-
-# Copy your list from above, and in this section
-# determine what data structure each layer should have
 
-########################################################
-# Step 3: Make the data structure!
+# - the number of rides each driver has given
+puts "Number of rides:"
+ride_share.each_pair { |id, rides| puts "Driver: #{id} ==> #{rides.length} rides"  }
+driver_earning = {}
 
-# Setup the entire data structure:
-# based off of the notes you have above, create the
-# and manually write in data presented in rides.csv
-# You should be copying and pasting the literal data
-# into this data structure, such as "DR0004"
-# and "3rd Feb 2016" and "RD0022"
+# - the total amount of money each driver has made
+puts "\nDrivers' earning:"
+ride_share.each_pair do |id, rides|
+  earning = 0
+  rides.each {|ride| earning += ride[:cost]}
```

Comment:
You can clarify this by using `sum`:

```suggestion
  earning = rides.sum {|ride| ride[:cost]}
```